### ISSUE TITLE
Castle thumbnail missing "X" after building

### ISSUE THREAD

**Issue by jasonbtiller** [ihhub/fheroes2#7178]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Linux

### Describe the bug

In the castle building screen, after building the castle's thumbnail does not get the "X" in the right-hand corner indicating it can't have new building this turn. Only after leaving the castle overview and returning to the castle overview does the thumbnail get the "X" badge.

I am unable to attach files for some reason, so I can't provide images or a save file, but here's how I reproduce it:

1. Start a new game.
2. Open castle overview.
3. Click on castle to bring up build window.
4. Note that the castle thumbnail has no "X" in the right-hand corner.
5. Build something, like the statue.
6. You automatically return to the castle overview screen; note that the castle thumbnail has no "X" badge.
7. Exit the castle overview window to the adventure window - now the castle thumbnail updates properly.

I believe that the castle thumbnail should receive the "X" badge as soon as you build something without having to return to the adventure screen.

### Save file

I can't provide a save because Firefox isn't allowing me to attach files...

**Comment by oleg-derevenetz** [comment]
If you are talking about castle thumbnail on the adventure map screen, i.e. here:

![buttons](https://github.com/ihhub/fheroes2/assets/32623900/23fe2626-9bc5-440b-af28-522ce794becf)

then this is by design: no details of the adventure screen are updated (be it the state of castles, heroes, or kingdom resources) while any dialogs are open on top of it. For example, when you hire a hero in a castle, this hero also does not appear in the list of heroes on the adventure map screen until you close the dialog window. This is not a bug. At best, this issue can be considered as a low-priority improvement.

**Comment by jasonbtiller** [comment]
I concur, @oleg-derevenetz. After I was watching the behavior of my GoG HoMM2, I realized that this was the original behavior. Sorry to clutter up the issues list.

**Comment by ihhub** [comment]
The same as #2598.

**Comment by ihhub** [comment]
Closing as duplicate of #5976 .
